cmd/gossamer/config.go

```diff
@@ -287,115 +288,98 @@ func createExportConfig(ctx *cli.Context) (*dot.Config, error) {
-	if cfg == nil {
-		cfg = new(ctoml.Config)
-	}
-
-	if lvlStr := ctx.String(LogFlag.Name); lvlStr != "" {
-		if lvlToInt, err := strconv.Atoi(lvlStr); err == nil {
-			lvlStr = log.Lvl(lvlToInt).String()
-		}
-		cfg.Global.LogLvl = lvlStr
-	}
+type getStringer interface {
+	String(key string) (value string)
+}
 
-	if cfg.Global.LogLvl == "" {
-		cfg.Global.LogLvl = gssmr.DefaultLvl.String()
+// getLogLevel obtains the log level in the following order:
+// 1. Try to obtain it from the flag value corresponding to flagName.
+// 2. Try to obtain it from the TOML value given, if step 1. failed.
+// 3. Return the default value given if both previous steps failed.
+// For steps 1 and 2, it tries to parse the level as an integer to convert it
+// to a level, and also tries to parse it as a string.
+func getLogLevel(ctx getStringer, flagName, tomlValue string, defaultLevel log.Lvl) (
+	level log.Lvl, err error) {
+	if flagValue := ctx.String(flagName); flagValue != "" {
+		return parseLogLevelString(flagValue)
 	}
 
-	var err error
-	globalCfg.LogLvl, err = log.LvlFromString(cfg.Global.LogLvl)
-	if err != nil {
-		return err
+	if tomlValue == "" {
+		return defaultLevel, nil
 	}
 
-	// check and set log levels for each pkg
-	if cfg.Log.CoreLvl == "" {
-		logCfg.CoreLvl = globalCfg.LogLvl
-	} else {
-		lvl, err := log.LvlFromString(cfg.Log.CoreLvl)
-		if err != nil {
-			return err
-		}
+	return parseLogLevelString(tomlValue)
+}
 
-		logCfg.CoreLvl = lvl
-	}
+var ErrLogLevelIntegerOutOfRange = errors.New("log level integer can only be between 0 and 5 included")
 
-	if cfg.Log.SyncLvl == "" {
-		logCfg.SyncLvl = globalCfg.LogLvl
-	} else {
-		lvl, err := log.LvlFromString(cfg.Log.SyncLvl)
-		if err != nil {
-			return err
+func parseLogLevelString(logLevelString string) (logLevel log.Lvl, err error) {
+	levelInt, err := strconv.Atoi(logLevelString)
+	if err == nil { // level given as an integer
+		if levelInt < 0 || levelInt > 5 {
+			return 0, fmt.Errorf("%w: log level given: %d", ErrLogLevelIntegerOutOfRange, levelInt)
 		}
+		logLevel = log.Lvl(levelInt)
+		return logLevel, nil
+	}
 
-		logCfg.SyncLvl = lvl
+	logLevel, err = log.LvlFromString(logLevelString)
+	if err != nil {
+		return 0, fmt.Errorf("cannot parse log level string: %w", err)
 	}
 
-	if cfg.Log.NetworkLvl == "" {
-		logCfg.NetworkLvl = globalCfg.LogLvl
-	} else {
-		lvl, err := log.LvlFromString(cfg.Log.NetworkLvl)
-		if err != nil {
-			return err
-		}
+	return logLevel, nil
+}
 
-		logCfg.NetworkLvl = lvl
+func setLogConfig(ctx getStringer, cfg *ctoml.Config, globalCfg *dot.GlobalConfig, logCfg *dot.LogConfig) (err error) {
+	if cfg == nil {
+		cfg = new(ctoml.Config)
 	}
 
-	if cfg.Log.RPCLvl == "" {
-		logCfg.RPCLvl = globalCfg.LogLvl
-	} else {
-		lvl, err := log.LvlFromString(cfg.Log.RPCLvl)
-		if err != nil {
-			return err
-		}
-
-		logCfg.RPCLvl = lvl
+	globalCfg.LogLvl, err = getLogLevel(ctx, LogFlag.Name, cfg.Global.LogLvl, gssmr.DefaultLvl)
```

Comment:
a suggestion using a list of parameters:
```suggestion
        toDefineLvls := []struct {
		lvl       *log.Lvl
		flagName  string
		tomlValue string
		def       log.Lvl
	}{
		{
			lvl:       &globalCfg.LogLvl,
			flagName:  LogFlag.Name,
			tomlValue: cfg.Global.LogLvl,
			def:       gssmr.DefaultLvl,
		},
		...
	}

	for _, l := range toDefineLvls {
		gotLvl, err := getLogLevel(ctx, l.flagName, l.tomlValue, l.def)
		if err != nil {
			return fmt.Errorf("cannot get %s: %w", l.flagName, err)
		}
		*l.lvl = gotLvl
	}
```

Reply:
Done :+1: Nice suggestion, I never thought about doing it like that!